vizro-core/src/vizro/models/_action/_action.py

```diff
@@ -146,25 +152,40 @@ def build(self):
         Returns:
             List of required components (e.g. dcc.Download) for the Action model added to the `Dashboard` container.
         """
-        callback_inputs, callback_outputs, action_components = self._get_callback_mapping()
+        external_callback_inputs, external_callback_outputs, action_components = self._get_callback_mapping()
+        callback_inputs = {
+            "external": external_callback_inputs,
+            "internal": {"trigger": Input({"type": "action_trigger", "action_name": self.id}, "data")},
+        }
+        callback_outputs = {
+            "internal": {"action_finished": Output("action_finished", "data", allow_duplicate=True)},
+        }
+
+        # If there are no outputs then we don't want the external part of callback_outputs to exist at all.
+        # This allows the action function to return None and match correctly on to the callback_outputs dictionary
+        # The (probably better) alternative to this would be just to define a dummy output for all such functions
+        # so that the external key always exists.
```

Comment:
@petar-qb any thoughts on this idea. I think options are:
1. you think it's a huge improvement _and_ it's quick and easy to do ➡️ let's do it here
2. you think it's a small improvement or it's not quick and easy to do ➡️ let's do it but not before release
3. you think it's not an improvement at all ➡️ let's not do it at all and remove the comment

Reply:
Excellent, let's leave it for a future change then. Please will you make sure it's captured somewhere? (Maybe that long list of planned improvements to the action implementation?)